Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,411 @@
+        {
+            var match = _context.Match
+                        .Where(m => m.SeriesID == seriesID)
+                        .Select(m => new MatchExtendedViewModel
+                        {
+                            ID = m.ID,
+                            Time = m.Time,
+                            Surface = _context.Surface
+                                        .FirstOrDefault(s => s.ID == m.SurfaceID)
+                                        .Description,
+                            Status = _context.MatchStatus
+                                        .FirstOrDefault(ms => ms.ID == m.StatusID)
+                                        .Description,
+                            Team = m.MatchTeam.Select(mt => new TeamExtendedViewModel
+                            {
+                                ID = mt.TeamID,
+                                Name = _context.Team
+                                   .FirstOrDefault(t => t.ID == mt.TeamID)
+                                   .Name,
+                                TeamRecord = _context.SeriesTeam
+                                           .Where(st => st.SeriesID == m.SeriesID && st.TeamID == mt.TeamID)
+                                           .Select(st => new TeamRecordViewModel
+                                           {
+                                               Win = st.Win,
+                                               Loss = st.Loss,
+                                           })
+                            })
+                        });
+            return match;
+        }
+        public async Task<MatchViewModel> PostMatchAsync(MatchUploadViewModel m)
+        {
+            var match = new Match
+            {
+                SeriesID = m.SeriesID,
+                Time = m.StartTime,
+                SurfaceID = m.SurfaceID ?? 1, //unknown surface id
+                StatusID = 1 //default unconfirmed
+            };
+            await _context.Match.AddAsync(match);
+            await _context.SaveChangesAsync();
+
+            foreach (var teamID in m.TeamIDs)
+            {
+                await CreateMatchTeamMappingAsync(teamID, match.ID);
+            }
+            await _context.SaveChangesAsync();
+            return match;
+        }
+
+        public async Task CreateMatchTeamMappingAsync(int teamID, int matchID)
+        {
+            var matchTeam = new MatchTeam
+            {
+                TeamID = teamID,
+                MatchID = matchID,
+                StatusID = 2, //default confirmed
+                Score = 0,
+                Sportsmanship = 5 //default max
+            };
+            await _context.MatchTeam.AddAsync(matchTeam);
+        }
+        public async Task<MatchTeamViewModel> UpdateTeamStatsAsync(int matchID, MatchStatsPatchViewModel vm)
+        {
+            var teamstats = _context.MatchTeam.FirstOrDefault(mt => mt.MatchID == matchID && mt.TeamID == vm.TeamID);
```

Comment:
You can also use `Find` in situations like this where there are multiple keys. Check the docs for specifications and the note on ordering: https://learn.microsoft.com/en-us/ef/ef6/querying/?source=recommendations#finding-an-entity-by-composite-primary-key

Reply:
Interesting however, I don't actually think we have any specific constraints on our DB that states that our join table's FK's can act as a composite PK. So most of these queries will be necessary unless we do a DB rework.